pulsar4s-avro/src/main/scala/com/sksamuel/pulsar4s/avro/package.scala

```diff
@@ -0,0 +1,61 @@
+package com.sksamuel.pulsar4s
+
+import java.io.ByteArrayOutputStream
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets
+
+import com.sksamuel.avro4s.AvroSchema
+import com.sksamuel.avro4s.AvroInputStream
+import com.sksamuel.avro4s.AvroOutputStream
+import com.sksamuel.avro4s.Decoder
+import com.sksamuel.avro4s.Encoder
+import com.sksamuel.avro4s.SchemaFor
+import org.apache.pulsar.client.api.Schema
+import org.apache.pulsar.common.schema.{SchemaInfo, SchemaType}
+
+import scala.annotation.implicitNotFound
+
+/**
+  * Automatic Schema derivation using avro4s
+  *
+  * == Examples ==
+  *
+  * {{{
+  *  import com.sksamuel.pulsar4s.avro._
+  *
+  *  case class City(id: Int, name: String)
+  *
+  *  producer.send(City(1, "London"))
+  *
+  *  val city: City = consumer.receive
+  *
+  * }}}
+  */
+package object avro {
+
+  @implicitNotFound("No Avro Schema for type ${T} found.")
+  implicit def avroSchema[T: Manifest: SchemaFor: Encoder: Decoder]: Schema[T] = new Schema[T] {
```

Comment:
@sksamuel Mind taking a quick look before I merge this? I think this looks right, but it feels like the implementation could be simpler.

Reply:
This looks about as simple as you can go at the moment. I think perhaps a way to encode a value as a byte array directly in avro4s might make sense, then this implementation becomes a lot smaller.

Reply:
@gmethvin let's merge and release then?

Reply:
I don't have permissions to make releases but @sksamuel can.

Reply:
Are you ready for a release @gmethvin  ?